src/libraries/Native/Unix/System.Native/pal_random.c

```diff
@@ -63,11 +63,24 @@ Return 0 on success, -1 on failure.
 */
 int32_t SystemNative_GetCryptographicallySecureRandomBytes(uint8_t* buffer, int32_t bufferLength)
 {
+    assert(buffer != NULL);
+
+#ifdef __EMSCRIPTEN__
+    extern int32_t dotnet_browser_entropy(uint8_t* buffer, int32_t bufferLength);
+    static bool sMissingBrowserCrypto;
```

Comment:
I think the caching is redundant

Reply:
It follows what is already in place and keeps us from calling external routine again unless I am missing something.  When we were aborting from the js code I would agree but recent mod now returns the -1 from the routine which will allow the managed exception be called.

Let me know if this needs to changed.